lib/iris/_constraints.py

```diff
@@ -490,3 +493,107 @@ def _cube_func(self, cube):
 
     def __repr__(self):
         return "AttributeConstraint(%r)" % self._attributes
+
+
+class NameConstraint(Constraint):
+    """Provides a simple Cube name based :class:`Constraint`."""
+
+    def __init__(
+        self,
+        standard_name="none",
+        long_name="none",
+        var_name="none",
+        STASH="none",
+    ):
+        """
+        Provides a simple Cube name based :class:`Constraint`, which matches
+        against each of the names provided, which may be either standard name,
+        long name, NetCDF variable name and/or the STASH from the attributes
+        dictionary.
+
+        The name constraint will only succeed if *all* of the provided names
+        match.
+
+        Kwargs:
+        * standard_name:
+            A string or callable representing the standard name to match
+            against.
+        * long_name:
+            A string or callable representing the long name to match against.
+        * var_name:
+            A string or callable representing the NetCDF variable name to match
+            against.
+        * STASH:
+            A string or callable representing the UM STASH code to match
+            against.
+
+        .. note::
+            The default value of each of the keyword arguments is the string
+            "none", rather than the singleton None, as None may be a legitimate
+            value to be matched against e.g., to constrain against all cubes
+            where the standard_name is not set, then use standard_name=None.
+
+        ... note::
+            The None value will not be passed through to a callable. Instead
```

Comment:
Unfortunately I really didn't find this clear.
I think I do get what it does now + why, but probably only because of the code comment below.
So I tried to explain the purpose + behaviour more clearly myself, and came up with.. 
```
    If a keyword argument is a callable, it is only called for cubes with the
    required name set.  So the callable is only ever passed a name, not a
    'None'.  All cubes without the required name are discarded.
```

Can you maybe improve, something along those lines ?

Reply:
@pp-mo Perhaps it's a level of detail that's just not necessary for the user to know.

That's a valid stance to take. Publicising this, no matter how well worded, is courting confusion.

That said, would you be adverse to me removing this `note` from the doc-string?